### PR TITLE
Mhp 2098 use no response filter

### PR DESCRIPTION
**Pull request by jesusfreak3**

### CODE REVIEW

File: app/assets/javascripts/angular/components/peopleFiltersPanel/peopleFiltersPanel.html [hunk 1]
```diff
@@ -126,17 +126,20 @@
                     <label class="filter-option flex-column" ng-if="question.kind === 'TextField'">
                         <select
                             class="w-100"
+                            ng-init="$ctrl.filters.answerMatchingOptions[question.id] = 'contains'"
                             ng-model="$ctrl.filters.answerMatchingOptions[question.id]"
-                            ng-change="$ctrl.updateFilters()"
-                            ng-init="$ctrl.filters.answerMatchingOptions[question.id] = 'contains'">
+                            ng-change="$ctrl.updateFilters($ctrl.updateFilters($ctrl.filters.answerMatchingOptions[question.id], question.id))">
```
Comment (OzzieOrca) [inline app/assets/javascripts/angular/components/peopleFiltersPanel/peopleFiltersPanel.html]: Why is `$ctrl.updateFilters` called twice?

File: app/assets/javascripts/angular/components/peopleFiltersPanel/peopleFiltersPanel.component.js [hunk 2]
```diff
@@ -83,7 +83,18 @@ function peopleFiltersPanelController(
...
 
-    this.updateFilters = () => {
+    this.updateFilters = (option, questionId) => {
+        const questionFilter =
+            option === 'no_response' || option === 'any_response' ? option : '';
+
+        if (questionId) {
+            const questions = this.filters.questions;
+            this.filters.questions = {
+                ...questions,
```
Comment (OzzieOrca) [inline app/assets/javascripts/angular/components/peopleFiltersPanel/peopleFiltersPanel.component.js]: I don't think you need the `questions` variable.
```suggestion
                ...this.filters.questions,
```

File: app/assets/javascripts/angular/components/peopleFiltersPanel/peopleFiltersPanel.html (hunk 1 above)
Comment (jesusfreak3) [reply app/assets/javascripts/angular/components/peopleFiltersPanel/peopleFiltersPanel.html]: not sure, its how it has always been, they define a init, which i guess triggers the ng change?  But the ng initi is needed to set the default option for each question?

Comment (OzzieOrca) [reply app/assets/javascripts/angular/components/peopleFiltersPanel/peopleFiltersPanel.html]: No, literally why is it `ng-change="$ctrl.updateFilters($ctrl.updateFilters(`? `updateFilters` is nested inside `updateFilters`.

File: app/assets/javascripts/angular/components/peopleFiltersPanel/peopleFiltersPanel.component.js (hunk 2 above)
Comment (jesusfreak3) [reply app/assets/javascripts/angular/components/peopleFiltersPanel/peopleFiltersPanel.component.js]: yup, true, should have cleaned that up, was attempting another way before this version.

File: app/assets/javascripts/angular/components/peopleFiltersPanel/peopleFiltersPanel.html (hunk 1 above)
Comment (jesusfreak3) [reply app/assets/javascripts/angular/components/peopleFiltersPanel/peopleFiltersPanel.html]: ahaha... oh... wow.. i didnt even see that.. thats incredible...

**Pull request merged**